assignment2/syntax.in

```diff
@@ -0,0 +1,30 @@
+<program> ::= <compoundstmt>
+
+<compoundstmt> ::= begin <stmtlist> end
+<stmtlist> ::= <statement> | <statement>; <stmtlist>
+<statement> ::= <assignstmt> | <declarationstmt> | <iostmt> | <ifstmt> | <forstmt> | <whilestmt> | <returnstmt> 
+
+<assignstmt> ::= <identifier>=<expression>
+<expression> ::= <expression> + <term> | <expression> - <term> | <term>
+<term> ::= <term> * <factor> | <term> / <factor> | <factor>
+<factor> ::= (<expression>) | <identifier> | <integer_const> | <array_factor_const> | {<integer_const>} | {<factor> <integer_const}
```

Comment:
arrays should be treated differently as opposed to simple data structures like numbers